hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlUnionTest.java

```diff
@@ -240,19 +242,122 @@ public void watermarkedStreamUnionAllTest() {
         ResultIterator<SqlRow> iterator = (ResultIterator<SqlRow>) result.iterator();
         List<Row> actualRows = new ArrayList<>();
         assertTrueEventually(() -> {
-            while (iterator.hasNext(50, TimeUnit.MILLISECONDS) == HasNextResult.YES) {
-                actualRows.add(new Row(iterator.next()));
-            }
-            if (actualRows.size() == 7 && expected.size() < 7) {
-                // Because we drop late items after union all (thanks to UnionDropLateItemsTransposeRule),
-                // it can happen that one of the late items makes it through. But both will never make it.
-                expected.add(new Row(timestampTz(0L), 4));
-            }
-            assertThat(actualRows).containsExactlyInAnyOrderElementsOf(expected);
-        },
+                    while (iterator.hasNext(50, TimeUnit.MILLISECONDS) == HasNextResult.YES) {
+                        actualRows.add(new Row(iterator.next()));
+                    }
+                    if (actualRows.size() == 7 && expected.size() < 7) {
+                        // Because we drop late items after union all (thanks to UnionDropLateItemsTransposeRule),
+                        // it can happen that one of the late items makes it through. But both will never make it.
+                        expected.add(new Row(timestampTz(0L), 4));
+                    }
+                    assertThat(actualRows).containsExactlyInAnyOrderElementsOf(expected);
+                },
                 5);
     }
 
+    @Test
+    public void prunableUnionAllTest() {
```

Comment:
we have other such tests in `PartitionPruningIT`. How do you plan structure the tests? Doing it in 2 different ways makes it harder to find them.

Reply:
cc @ivanthescientist

Reply:
@ivanthescientist do you think that this is correct place for these tests?